src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -482,6 +477,12 @@ public List<PackageManagerSearchElement> ListAll()
         {
             CachedPackageList = new List<PackageManagerSearchElement>();
 
+            // Attempt to load user votes prior to using it
+            if (AuthenticationManager.LoginState.Equals(LoginState.LoggedIn) && Uservotes == null)
+            {
+                Task.Run(() => this.Uservotes = this.Model.UserVotes());
```

Comment:
also Task.Run() will run asynchronously on another thread. That means that the rest of the code will execute without waiting for the result.
Also won't `Model.ListAll()` cause the same UI blocking issue ?

Ideally we should make all these methods async and cascade them down to a UI method that makes sense to be async. Something like DynamoViewModelRequestShowPackageManagerSearch should be async (and all the async-able calls downstream) and the UI should relate that it is doing something async (UI spinner etc). I realize this is too much for 3.0.x release.

I am no longer sure how we can make this work properly, since a part of the code needs the information synchronously (which means the UI thread needs to wait for that information)

Reply:
OF course, just by moving the offending outside the constructor, you will no longer block the DynamoView initialization code (basically Dynamo load logic). We will now block the UI, but later on when the user actually requests PackageManager information.
Maybe we it is better than nothing.

We could also try to parallelize the calls something like
```
           IEnumerable<PackageHeader> allPackages;
           var task1 = Task.Run(() => Model.UserVotes());
           var task2 = Task.Run(() => Model.ListAll());

           Task.WhenAll(task1, task2).Wait();

           Uservotes = task1.Result;
           allPackages = task2.Result;
```

Reply:
Sorry, looks like there are some unresolved discussions around REstSHarp's thread safety.
I think it is ok to call it from multiple threads, just not in parallel
So, runnig the code on the UI thread might be safer (There is no real value in using Task.Run except if you can do it async or parallelize it)